src/protocols/secure_channel/StatusReport.cpp

```diff
@@ -77,8 +78,7 @@ CHIP_ERROR StatusReport::Parse(System::PacketBufferHandle buf)
 
 Encoding::LittleEndian::BufferWriter & StatusReport::WriteToBuffer(Encoding::LittleEndian::BufferWriter & buf) const
 {
-    static_assert(std::is_same<std::underlying_type_t<decltype(mGeneralCode)>, uint16_t>::value, "Cast is not safe");
```

Comment:
Shouldn't we leave the static_asserts in such cases where we expect the underlying type is uint16_t etc?. `to_underlying` doesn't provide us this check, right?

Reply:
It does not, but the check was only needed due to the manual casting we are doing.  With the use of `to_underlying` there are two things that can happen with `Put16(to_underlying(mGeneralCode))`:

1. The underlying type fits in `uint16_t`.  In that case, everything is fine.
2. The underlying type does not fit in `uint16_t`.  In that case our `-Wconversion` warnings will trigger and compilation will fail.

We _could_ keep some form of the static assert to make the second case not depend on warning compilation settings.  But it's a fair amount of extra noise and the whole point of enabling `-Wconversion` is to not have to manually write out all the checks the compiler can just do....

Reply:
Ok, makes sense. Thanks!